content/blog/2025/size-preferences.md

```diff
@@ -0,0 +1,539 @@
+
+For my eyes, on my laptop,
+a roughly `20px` font-size feels better
+than the `16px` browser default.
+So it seems to me that I should set my
+user preference to `20px`,
+and that change should
+improve my experience of the web.
+
+I don't expect that I will see
+_exactly_ `20px` fonts for everything --
+and I wouldn't want to!
+
+- **Not all content has the same needs.**
+  When I'm reading a blog post
+  I like larger text,
+  but when I'm filling in a spreadsheet
+  smaller text makes sense --
+  as long as I can still read it.
+  There's no preference setting
+  that will adapt perfectly in both situations.
+- **All styles are contextual**,
+  and it's the overall impact that we're concerned with.
+  My `16px` default size will look different
+  depending on the fonts used,
+  the space available,
+  the color scheme, line-lengths, white-space,
+  and so on.
+- **Fluid type is (by definition) a dynamic value**.
+  If we do want to integrate viewport units later,
+  we can't do that while also matching
+  a user preference _exactly_.
+
+So global preference are limited,
+and will never provide a perfect solution.
+We need some way for authors to adapt those preferences
+to be more appropriate for a given context,
+and a page-zoom option for users
+to make final adjustment on a site-by-site basis.
+
+But the global setting should mean _something_,
+and ideally it's a value users are comfortable changing
+to match their _actual preference_.
+So the question for us as authors is
+how best to interpret and adapt that preference
+so that it is useful on the sites we build,
+even when it's not applied _directly_.
+
+[ultimate]: https://adrianroselli.com/2024/03/the-ultimate-ideal-bestest-base-font-size-that-everyone-is-keeping-a-secret-especially-chet.html
+
+### The current approach multiplies
+
+Current best practice
+starts with selecting an ideal pixel-based font-size
+that we like for our site design,
+and converting that to `em`s or `rem`s
+by assuming `1em = 16px` --
+the common default provided by browsers.
+The result can be applied
+on the `html` element with `em`
+or on the `body` tag with `rem`.
+Often,
+this conversion is done in third-party tools,
+but we can also do the math explicitly in CSS:
+
+```css
+html {
+  /* this should work with pixel units, but browsers don't support it yet */
+  --DEFAULT: 16;
+  --ideal: 24;
+
+  /* calc(24 / 16 * 1rem) */
+  font-size: calc(var(--ideal) / var(--DEFAULT) * 1em);
+}
+```
+
+Using that approach,
+we achieve two important goals:
+
+1. We (as authors) get to choose
+   an ideal base size for our site content
+   (in this case `24px` with units removed),
+   and it will apply for the majority of users
+   who never change their default setting
+2. When users do change their font-size preference,
+   our site text scales up or down
+   in relation to the preference
+3. Users can also zoom in or out after they land on the page,
+   using built-in browser zoom tools
```

Comment:
```suggestion
1. We (as authors) get to choose
   an ideal base size for our site content
   (in this case `24px` with units removed),
   and it will apply for the majority of users
   who never change their default setting.
2. When users do change their font-size preference,
   our site text scales up or down
   in relation to the preference.
3. Users can also zoom in or out after they land on the page,
   using built-in browser zoom tools.
```